internal/service/meter.go

```diff
@@ -63,3 +64,75 @@ func (s *meterService) DisableMeter(ctx context.Context, id string) error {
 	}
 	return s.meterRepo.DisableMeter(ctx, id)
 }
+
+// contains checks if a slice contains a specific value
+func contains(slice []string, value string) bool {
+	for _, v := range slice {
+		if v == value {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *meterService) UpdateMeter(ctx context.Context, id string, filters []meter.Filter) (*meter.Meter, error) {
+	// Validate input
+	if id == "" {
+		return nil, fmt.Errorf("id is required")
+	}
+
+	if len(filters) == 0 {
+		return nil, fmt.Errorf("filters cannot be empty")
+	}
+
+	// Fetch the existing meter
+	existingMeter, err := s.meterRepo.GetMeter(ctx, id)
+	if err != nil {
+		return nil, fmt.Errorf("fetch meter: %w", err)
+	}
+
+	// Merge filters
+	mergedFilters := mergeFilters(existingMeter.Filters, filters)
+
+	// Update only the filters field in the database
+	if err := s.meterRepo.UpdateMeter(ctx, id, mergedFilters); err != nil {
+		return nil, fmt.Errorf("update filters: %w", err)
+	}
+
+	// Return the updated meter object
+	existingMeter.Filters = mergedFilters
+	return existingMeter, nil
```

Comment:
first update the existingMeter object
`existingMeter.Filters = mergedFilters`
then pass the entire meter object to `s.meterRepo.UpdateMeter(ctx, existingMeter)`